### PR TITLE
[localize-tools] Convert localize-tools to JS modules and tape

### PR DESCRIPTION
**Pull request by aomarks**
- Convert from CommonJS to JS modules
- Convert from ava to tape (using https://github.com/vanillaes/tape-es as test runner)
- Refactor config types because `typescript-json-schema` doesn't work properly with JS modules. Also now runs much faster as a side effect.

### CODE REVIEW

File: packages/localize-tools/src/cli.ts [hunk 1]
```diff
@@ -72,9 +75,16 @@ export async function runAndLog(argv: string[]): Promise<number> {
   return 0;
 }
 
-function version() {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  return require(path.join('..', 'package.json')).version;
+async function version() {
+  const packageJsonPath = path.resolve(
+    dirname(fileURLToPath(import.meta.url)),
+    '..',
```
Comment (justinfagnani) [inline packages/localize-tools/src/cli.ts:81]: with resolve() you can do `'../package.json'`

File: packages/localize-tools/src/cli.ts [hunk 2]
```diff
@@ -72,9 +75,16 @@ export async function runAndLog(argv: string[]): Promise<number> {
   return 0;
 }
 
-function version() {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  return require(path.join('..', 'package.json')).version;
+async function version() {
+  const packageJsonPath = path.resolve(
+    dirname(fileURLToPath(import.meta.url)),
```
Comment (justinfagnani) [inline packages/localize-tools/src/cli.ts:80]: I don't think you need `dirname()`

File: packages/localize-tools/src/patches.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {Locale} from './locales';
+import {Locale} from './types/locale';
```
Comment (justinfagnani) [inline packages/localize-tools/src/patches.ts:7]: add `.js`?

**Review by justinfagnani — APPROVED** [review]

File: packages/localize-tools/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Changed
+
 - Initial release of `@lit/localize-tools` package. This new package provides
   the `lit-localize` binary, while `@lit/localize` continues to provide the
   browser library (`msg`, `LocalizedElement`, etc.).
+
+- **BREAKING** `lit-localize` now uses JS modules instead of CommonJS, so it
```
Comment (rictic) [inline packages/localize-tools/CHANGELOG.md:16]: hm. internally we're still on Node 12

Comment (aomarks) [reply packages/localize-tools/CHANGELOG.md:16]: Internally we'd be consuming the TypeScript directly though, so the way the NPM binary is built doesn't have any effect right?

Comment (rictic) [reply packages/localize-tools/CHANGELOG.md:16]: oh, true!

File: packages/localize-tools/src/cli.ts [hunk 5]
```diff
@@ -5,16 +5,19 @@
...
-import {TransformLitLocalizer, TransformOutputConfig} from './modes/transform';
-import {RuntimeLitLocalizer, RuntimeOutputConfig} from './modes/runtime';
-
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-require('source-map-support').install();
+import {TransformLitLocalizer} from './modes/transform.js';
+import {RuntimeLitLocalizer} from './modes/runtime.js';
+import {TransformOutputConfig, RuntimeOutputConfig} from './types/modes.js';
+import {dirname} from 'path';
+import {fileURLToPath} from 'url';
```
Comment (rictic) [inline packages/localize-tools/src/cli.ts:18]: oh nice!

File: packages/localize-tools/src/locales.ts [hunk 6]
```diff
@@ -4,12 +4,11 @@
...
 
-import * as fsExtra from 'fs-extra';
+import fsExtra from 'fs-extra';
 import * as pathLib from 'path';
-import {KnownError} from './error';
-import {escapeStringToEmbedInTemplateLiteral} from './typescript';
-
-export type Locale = string & {__TYPE__: 'Locale'};
+import {KnownError} from './error.js';
+import {Locale} from './types/locale';
```
Comment (rictic) [inline packages/localize-tools/src/locales.ts:10]: Any reason to leave off the .js here?

File: packages/localize-tools/src/tests/analysis.unit.test.ts [hunk 7]
```diff
@@ -64,6 +65,7 @@ function checkAnalysis(
 test('irrelevant code', (t) => {
   const src = 'const foo = "foo";';
   checkAnalysis(t, src, []);
+  t.end();
```
Comment (rictic) [inline packages/localize-tools/src/tests/analysis.unit.test.ts:68]: hm, the requirement for an explicit t.end() is awkward and accident prone compared to just awaiting the return value of the test

wrapper function?

File: packages/localize-tools/src/locales.ts (hunk 6 above)
Comment (rictic) [reply packages/localize-tools/src/locales.ts:10]: Oh I see, it's a .d.ts file, right. Consider `import type` for clarity?

**Review by rictic — APPROVED** [review]

File: packages/localize-tools/src/patches.ts (hunk 3 above)
Comment (aomarks) [reply packages/localize-tools/src/patches.ts]: Done.

**Pull request merged**